plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py

```diff
@@ -144,18 +140,18 @@ def test_optuna_example(with_commandline: bool, tmpdir: Path) -> None:
         ]
     run_python_script(cmd)
     returns = OmegaConf.load(f"{tmpdir}/optimization_results.yaml")
+    study = optuna.load_study(storage=storage, study_name=study_name)
+    best_trial = study.best_trial
     assert isinstance(returns, DictConfig)
     assert returns.name == "optuna"
-    assert returns["best_params"]["x"] == 0
+    assert returns["best_params"]["x"] == best_trial.params["x"]
     if with_commandline:
         assert "y" not in returns["best_params"]
     else:
-        assert returns["best_params"]["y"] == 0
-    assert returns["best_value"] == 0
+        assert returns["best_params"]["y"] == best_trial.params["y"]
+    assert returns["best_value"] == best_trial.value
```

Comment:
Can we use constants here (with a reasonable epsilon?)
The current testing looks like it will be too lenient to  functional issues in Optuna.

Reply:
Thank you for your comment.

I understand your concern since the current Optuna repository does not have CI jobs which execute search performance tests.

I tried to estimated the reasonable epsilon with [a notebook](https://colab.research.google.com/drive/1HIvjys_NhinskSR1AFlby0NsLstRlZUD?usp=sharing). I repeated 1000 time of random search (that is the search algorithm used in this example), and found that the 95th percentile value of the best values was 6.25.  

The acceptance range is shown in the following contour plot with the red rectangle. I guess it is reasonably small considering that the search algorithm was random search. What do you think?

![image](https://user-images.githubusercontent.com/3255979/128047934-bcf6439f-b400-4941-8335-0a2dbc448869.png)

The suggested change is as follows:
```python
    # 95th percentile with 1000 different seed values.
    assert returns["best_value"] <= 6.25
```

Reply:
Thanks for evaluating 
Seems like a very large range (and also, if I understand correctly, 95th percentile means it will fail with one in 20 random seeds).

Is there an algorithm that converges faster than a random search we can test with?

Reply:
I applied the following three changes to the test case.
1. Employ 99the percentile (it will fail with one in 100 random seeds.)
2. Use TPE sampler instead of the random search
3. Increase the number of trials from 20 to 40.

The threshold values were changed as follows:
1. `6.25` to `12.25`
2. `12.25` to `2.27`
3. `2.27` to `0.25`

So, I applied all changes in the commit https://github.com/facebookresearch/hydra/pull/1746/commits/268cbb7ae399d611c5e88e38b0dadbfb200e7886.

The change increased the computation time, but it was still shorter than that of random search with 100 trials.
```
random with 20 trials
2.90s call     plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py::test_optuna_example[False]
2.85s call     plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py::test_optuna_example[True]

random with 100 trials (from #1726)
8.89s call     plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py::test_optuna_example[False]
8.79s call     plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py::test_optuna_example[True]

tpe with 20 trials
2.97s call     plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py::test_optuna_example[False]
2.95s call     plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py::test_optuna_example[True]

tpe with 40 trials
5.15s call     plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py::test_optuna_example[False]
4.82s call     plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py::test_optuna_example[True]
```

Reply:
Given that the seed is fixed in the test, I think 95th percentile is acceptable.
What bounds do we get for 95th percentile, tpe and 20 iterations?

Go ahead and proceed with whatever values you are comfortable with after this evaluation. I think it will be a good idea to add a comment to this discussion to the test.

Reply:
Thank you for your suggestion.
The 95th percentile with TPE and 20 iterations is 2.27.
I updated the bound and added a code comment on this discussion in this commit https://github.com/facebookresearch/hydra/pull/1746/commits/e463885adacf2713ddee97f3ef8d381633ff57ed